pymongo/pool.py

```diff
@@ -571,7 +572,9 @@ def hello_cmd(self):
         if self.opts.server_api:
             return SON([(HelloCompat.CMD, 1)])
         else:
-            return SON([(HelloCompat.LEGACY_CMD, 1)])
+            if self.hello_ok:
+                return SON([(HelloCompat.CMD, 1)])
+            return SON([(HelloCompat.LEGACY_CMD, 1), ('helloOk', True)])
```

Comment:
Is helloOk only supposed to be sent with the initial handshake? If so we should move it to `_ismaster()` under the `if performing_handshake` check.

Reply:
Thanks for catching that. Done!

Reply:
I brought up this issue in the spec PR and James says we should send helloOk on all isMaster commands, not just the initial handshake.